sdk/daml-script/test/daml/upgrades/Interfaces.daml

```diff
@@ -0,0 +1,237 @@
+  (alice, icid) <- setupAliceAndInterface
+  res <- alice `submit` createAndExerciseCmd (Caller with party = alice) (CallInterface with cid = icid)
+  res === "V2"
+
+interfaceChoicePackagePreferenceCommand : Test
+interfaceChoicePackagePreferenceCommand = test $ do
+  (alice, icid) <- setupAliceAndInterface
+  res <-
+    submitWithOptions
+      (actAs alice <> packagePreference [interfacesV1])
+      (exerciseCmd icid GetVersion)
+  res === "V1"
+
+interfaceChoicePackagePreferenceUpdate : Test
+interfaceChoicePackagePreferenceUpdate = test $ do
+  (alice, icid) <- setupAliceAndInterface
+  res <-
+    submitWithOptions
+      (actAs alice <> packagePreference [interfacesV1])
+      (createAndExerciseCmd (Caller with party = alice) (CallInterface with cid = icid))
+  res === "V1"
+
+multipleInterfaceChoicePackagePreference : (Party -> ContractId I -> Commands Text) -> Test
+multipleInterfaceChoicePackagePreference doExercise = test $ do
+  alice <- allocateParty "alice"
+  cid <- alice `submit` createExactCmd V1.FITemplate with party = alice
+  cidAlt <- alice `submit` createExactCmd V1.FITemplateAlt with party = alice
+  let icid = toInterfaceContractId @I cid
+      icidAlt = toInterfaceContractId @I cidAlt
+      -- Tests all 4 combinations of 2 versions for 2 packages
+      cases = 
+        [ (interfacesVersion, interfacesVersionStr, interfacesAltVersion, interfacesAltVersionStr)
+        | (interfacesVersion, interfacesVersionStr) <- [(interfacesV1, "V1"), (interfacesV2, "V2")]
+        , (interfacesAltVersion, interfacesAltVersionStr) <- [(interfacesAltV1, "V1"), (interfacesAltV2, "V2")]
+        ]
+
+  forM_ cases $ \(interfacesVersion, interfacesVersionStr, interfacesAltVersion, interfacesAltVersionStr) -> do
+    (res, resAlt) <-
+      submitWithOptions (actAs alice <> packagePreference [interfacesVersion, interfacesAltVersion]) $ do
+        res <- doExercise alice icid
+        resAlt <- doExercise alice icidAlt
+        pure (res, resAlt)
+    res === interfacesVersionStr
+    resAlt === interfacesAltVersionStr
+
+multipleInterfaceChoicePackagePreferenceCommand : Test
+multipleInterfaceChoicePackagePreferenceCommand =
+  multipleInterfaceChoicePackagePreference $ \_ icid ->
+    exerciseCmd icid GetVersion
+
+multipleInterfaceChoicePackagePreferenceUpdate : Test
+multipleInterfaceChoicePackagePreferenceUpdate =
+  multipleInterfaceChoicePackagePreference $ \alice icid ->
+    createAndExerciseCmd (Caller with party = alice) (CallInterface with cid = icid)
+
+queryInterfaceUpgrades : Test
+queryInterfaceUpgrades = test $ do
+  (alice, icid) <- setupAliceAndInterface
+  res <- fromSomeNote "Failed to find contract" <$> alice `queryInterfaceContractId` icid
+  res.version === "V2"
+
+fetchFromInterfaceUpgrades : Test
+fetchFromInterfaceUpgrades = test $ do
+  (alice, icid) <- setupAliceAndInterface
+  -- Using a v2 FITemplate to exercise the v1, to avoid adding another package that would need to depend on v2 FITemplate
```

Comment:
Maybe move this comment explaining the decision into the source of the files as well - having to read all the way through to here was fine for me because you explained it in our call, but would be confusing otherwise.